### PR TITLE
Improve student activity registration system

### PR DESCRIPTION
**Pull request by oksanatanaka**
This pull request adds participant management functionality to the High School Management System, allowing users to view, sign up for, and unregister from activities via both the backend API and frontend UI. It also introduces comprehensive tests to ensure the reliability of these features.

### Backend API enhancements

* Added new activities (`Basketball Team`, `Tennis Club`, `Art Studio`, `Music Band`, `Debate Club`, `Science Olympiad`) to the `activities` dictionary in `src/app.py`, expanding the available options for students.
* Implemented an `/activities/{activity_name}/unregister` POST endpoint in `src/app.py` to allow participants to be removed from activities, with error handling for non-existent activities and unregistered users.

### Frontend improvements

* Added the `unregisterParticipant` function in `src/static/app.js`, enabling users to remove participants from activities directly from the UI.
* Updated the activities list rendering in `src/static/app.js` to display participants for each activity, including a button to remove them.
* Ensured the activities list refreshes after signup or unregistration to keep the UI in sync with backend changes.
* Enhanced participant section styling in `src/static/styles.css` for improved readability and user experience.

### Testing

* Added `tests/test_api.py` with extensive tests for all API endpoints, including edge cases for signup and unregistration, and ensured data consistency between tests.
* Initialized the test package with `tests/__init__.py`.

### CODE REVIEW

File: src/app.py [hunk 1]
```diff
@@ -52,16 +88,29 @@ def get_activities():
     return activities
 
 
+
+from fastapi import Request
```
[the remark this answers is not in the document]
Comment (oksanatanaka) [reply src/app.py:92]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/oksanatanaka/skills-getting-started-with-github-copilot/pull/2#discussion_r2532374238)

File: src/app.py [hunk 2]
```diff
@@ -52,16 +88,29 @@ def get_activities():
...
-
-    # Add student
+    if email in activity["participants"]:
+        raise HTTPException(status_code=400, detail="Student already signed up for this activity")
     activity["participants"].append(email)
     return {"message": f"Signed up {email} for {activity_name}"}
+
+# Unregister endpoint
+@app.post("/activities/{activity_name}/unregister")
+async def unregister_participant(activity_name: str, request: Request):
```
[the remark this answers is not in the document]
Comment (oksanatanaka) [reply src/app.py:107]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/oksanatanaka/skills-getting-started-with-github-copilot/pull/2#discussion_r2532374286)

[the remark this answers is not in the document]
Comment (oksanatanaka) [reply src/app.py:107]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/oksanatanaka/skills-getting-started-with-github-copilot/pull/2#discussion_r2532374303)
